src/vivarium/component.py

```diff
@@ -697,15 +704,15 @@ def get_data(self, builder: Builder, data_source: DataInput) -> Any:
                     raise ConfigurationError(
                         f"There is no method '{method}' for the {module_string}."
                     )
-                data = data_source(builder)
+                data = data_source_callable(builder)
             else:
                 try:
                     data = builder.data.load(data_source)
                 except ArtifactException:
                     raise ConfigurationError(
                         f"Failed to find key '{data_source}' in artifact."
                     )
-        elif isinstance(data_source, Callable):
+        elif callable(data_source):
```

Comment:
today I learned about the built-in `callable` which made a bunch of mypy errors go away. Not sure why mypy didn't like `isinstance(data_source, Callable)`. It was throwing `src/vivarium/component.py:718: error: Argument 2 to "isinstance" has incompatible type "<typing special form>"; expected "_ClassInfo"  [arg-type]` as well a bunch of operation errors at the next line (the call itself). This one change got rid of all of them.